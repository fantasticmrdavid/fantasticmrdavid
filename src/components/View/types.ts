export type StateProps = {
  location: string,
  isLoading: boolean,
  isFirstLoad: boolean,
  popup: any,
};

export type DispatchProps = {
  updateLocation: (newLocation: string) => void,
};

export type ContainerProps = {
  location?: string,
  target?: string
};
