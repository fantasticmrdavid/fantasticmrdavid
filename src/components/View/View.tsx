import React, { memo, useState } from "react";
import { LogoMenu } from "components/LogoMenu/LogoMenu";
import { Home } from "views/Home/Home";
import { About } from "views/About/About";
import { Work } from "views/Work/Work";
import { Contact } from "views/Contact/Contact";
import GlobalStyles from "styles/global";
import { ImagePreloader } from "components/ImagePreloader/ImagePreloader";
import { LOCATIONS } from "routers/constants";
import {
  Container,
  Control,
  Media,
  ShutterTop,
  ShutterBottom,
  MediaLoadingMessage,
} from "./view.styles";

export type UiProps = {
  location: (typeof LOCATIONS)[keyof typeof LOCATIONS];
  target?: string;
};

export const View = memo(({ location, target }: UiProps) => {
  const [isLoading, setIsLoading] = useState(true);

  return (
    <>
      <ImagePreloader
        hideCompletedCount
        images={["/assets/images/legoFlinders.jpg"]}
        completedAction={() => setIsLoading(false)}
      />
      <GlobalStyles />
      <Container>
        <Control>
          <LogoMenu location={location} />
        </Control>

        <Media $location={location}>
          <ShutterTop $isFirstMediaLoading={isLoading} />
          <ShutterBottom $isFirstMediaLoading={isLoading} />
          <MediaLoadingMessage $isFirstMediaLoading={isLoading}>
            Hello!
          </MediaLoadingMessage>
          {location === LOCATIONS.home && <Home isLoading={isLoading} />}
          {location === LOCATIONS.about && <About isLoading={isLoading} />}
          {location === LOCATIONS.work && <Work target={target} />}
          {location === LOCATIONS.contact && <Contact />}
        </Media>
      </Container>
    </>
  );
});
