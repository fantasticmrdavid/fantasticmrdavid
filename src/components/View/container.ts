import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import ReactGA from 'react-ga';
import { AppState } from 'reducers';
import { locationUpdated } from './actions';
import Ui, { Props as UiProps } from './ui';

type StateProps = {
  location: string,
  loading: boolean,
  firstLoad: boolean,
  popup: any,
}

type DispatchProps = {
  updateLocation: (newLocation: string) => void,
}

type ContainerProps = {
  location?: string,
}

const mapStateToProps = (state: AppState, ownProps: ContainerProps) => {
  const { loading, popup } = state;
  const { location } = ownProps;
  return {
    location: state.location.current || location,
    loading: !location && loading.media,
    firstLoad: !location && loading.firstLoad,
    popup,
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>, ownProps: UiProps) => {
  return {
    updateLocation: (newLocation: string) => {
      const { location } = ownProps;
      const updatedLocation = location !== newLocation ? newLocation : 'home';

      dispatch(locationUpdated(updatedLocation));

      ReactGA.event({
        category: 'Navigation',
        action: 'Location Updated',
        label: updatedLocation,
      });
    },
  };
};

export default connect<StateProps, DispatchProps, ContainerProps, AppState>(
  mapStateToProps,
  mapDispatchToProps,
)(Ui);
