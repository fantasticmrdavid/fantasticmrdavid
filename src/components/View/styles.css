.View {
  display: flex;
  justify-content: center;
  align-items: stretch;
  min-height: 100%;
  background: $c-dark-charcoal;

  @media (max-width: $l-w-small) {
    display: block;
  }
}

.control {
  width: 40%;
  display: flex;
  justify-content: center;
  align-items: stretch;
  background: #232323;
  color: $c-text-primary;
  transition: 0.3s;

  @media (max-width: $l-w-small) {
    width: 100%;
  }

  @media (min-width: $l-w-medium) {
    min-height: 100vh;
  }
}

.media {
  position: relative;
  display: flex;
  width: 60%;

  @media (max-width: $l-w-small) {
    width: 100%;
  }

  &.work {
    height: 100vh;  /* IE */
  }
}

.mediaShutter {
  position: absolute;
  right: 0px;
  z-index: 16;
  width: 100%;
  height: 0%;
  transition: 0.3s;
  background: black;

  .firstMediaLoading & {
    height: 50%;
  }
}

.mediaShutterTop {
  composes: mediaShutter;
  top: 0px;
}

.mediaShutterBottom {
  composes: mediaShutter;
  bottom: 0px;
}

.mediaLoadingMessage {
  font-family: $ff-heading;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0px;
  right: 0px;
  justify-content: center;
  align-items: center;
  font-size: 2.5rem;
  font-weight: bold;
  color: white;
  z-index: 17;
  display: none;

  .firstMediaLoading & {
    display: flex;
  }
}
