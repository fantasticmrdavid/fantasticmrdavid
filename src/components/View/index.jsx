import React from 'react';
import PropTypes from 'prop-types';
import PopupContainer from 'containers/PopupContainer';
import LogoMenuContainer from 'containers/LogoMenuContainer';
import Home from 'views/Home';
import About from 'views/About';
import Work from 'views/Work';
import Contact from 'views/Contact';
import styles from './styles.css';

const View = ({ location, loading, firstLoad }) => {
  const firstMediaLoading = loading && firstLoad;
  return (
    <div className={`${styles.View} ${firstMediaLoading && styles.firstMediaLoading}`}>
      <PopupContainer />
      <section className={styles.control}>
        <LogoMenuContainer />
      </section>

      <section className={`${styles.media} ${styles[location]}`}>
        <div className={styles.mediaShutterTop} />
        <div className={styles.mediaShutterBottom} />
        <div className={styles.mediaLoadingMessage}>Hello!</div>
        { location === 'home' && <Home /> }
        { location === 'about' && <About loading={loading} /> }
        { location === 'work' && <Work /> }
        { location === 'contact' && <Contact /> }
      </section>
    </div>
  );
};

View.propTypes = {
  location: PropTypes.string.isRequired,
  loading: PropTypes.bool,
  firstLoad: PropTypes.bool,
};

export default View;
