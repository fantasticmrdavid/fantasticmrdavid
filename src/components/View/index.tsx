import React, { Fragment } from 'react';
import PopupContainer from 'containers/PopupContainer';
import LogoMenuContainer from 'containers/LogoMenuContainer';
import Home from 'views/Home';
import About from 'views/About';
import Work from 'views/Work';
import Contact from 'views/Contact';
import GlobalStyles from 'styles/global';
import * as styles from './styles';

export interface Props {
  location: string,
  loading: boolean,
  firstLoad: boolean,
}

export default ({ location, loading, firstLoad }: Props) => {
  const {
    Container,
    Control,
    Media,
    ShutterTop,
    ShutterBottom,
    MediaLoadingMessage,
  } = styles;
  const firstMediaLoading = loading && firstLoad;
  return (
    <Fragment>
      <GlobalStyles />
      <Container>
        <PopupContainer />
        <Control>
          <LogoMenuContainer />
        </Control>

        <Media location={location}>
          <ShutterTop firstMediaLoading={firstMediaLoading} />
          <ShutterBottom firstMediaLoading={firstMediaLoading} />
          <MediaLoadingMessage firstMediaLoading={firstMediaLoading}>Hello!</MediaLoadingMessage>
          { location === 'home' && <Home /> }
          { location === 'about' && <About loading={loading} /> }
          { location === 'work' && <Work /> }
          { location === 'contact' && <Contact /> }
        </Media>
      </Container>
    </Fragment>
  );
};
