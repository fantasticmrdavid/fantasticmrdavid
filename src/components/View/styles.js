import styled from 'styled-components';
import theme from 'styles/theme';
import { media } from 'styles/utils';

export const Container = styled.div`
  display: block;
  justify-content: center;
  align-items: stretch;
  min-height: 100%;
  background-color: ${theme.colorDarkCharcoal};

  ${media.small`
    display: flex;
  `}
`;

export const Control = styled.section`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: stretch;
  background: ${theme.colorSidebar};
  color: ${theme.colorTextPrimary};
  transition: 0.3s;

  ${media.small`
    width: 40%;
  `};

  ${media.medium`
    min-height: 100vh;
  `};
`;

export const Media = styled.section`
  position: relative;
  display: flex;
  width: 100%;
  height: ${props => (props.location === 'work' ? '100vh' : undefined)};

  ${media.small`
    width: 60%;
  `}
`;

const Shutter = styled.div`
  position: absolute;
  right: 0px;
  z-index: 16;
  width: 100%;
  height: ${props => (props.firstMediaLoading ? '50%' : '0%')};
  transition: 0.3s;
  background: ${theme.colorBlack};
`;

export const ShutterTop = styled(Shutter)`
  top: 0px;
`;

export const ShutterBottom = styled(Shutter)`
  bottom: 0px;
`;

export const MediaLoadingMessage = styled.div`
  display: ${props => (props.firstMediaLoading ? 'flex' : 'none')};
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0px;
  right: 0px;
  justify-content: center;
  align-items: center;
  font-family: ${theme.fontHeading};
  font-size: 2.5rem;
  font-weight: 700;
  color: ${theme.colorWhite};
  z-index: 17;
`;
