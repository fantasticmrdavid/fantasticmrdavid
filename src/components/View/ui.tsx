import React, { Fragment, memo } from 'react';
import LogoMenu from 'components/LogoMenu';
import Popup from 'components/Popup';
import Home from 'views/Home';
import About from 'views/About';
import Work from 'views/Work';
import Contact from 'views/Contact';
import GlobalStyles from 'styles/global';
import * as styles from './styles';

export interface Props {
  location: string,
  loading: boolean,
  firstLoad: boolean,
}

const {
  Container,
  Control,
  Media,
  ShutterTop,
  ShutterBottom,
  MediaLoadingMessage,
} = styles;

export default memo(
  ({ location, loading, firstLoad }: Props) => {
    const firstMediaLoading = loading && firstLoad;
    return (
      <Fragment>
        <GlobalStyles />
        <Container>
          <Popup />
          <Control>
            <LogoMenu location={location} />
          </Control>

          <Media location={location}>
            <ShutterTop firstMediaLoading={firstMediaLoading} />
            <ShutterBottom firstMediaLoading={firstMediaLoading} />
            <MediaLoadingMessage firstMediaLoading={firstMediaLoading}>Hello!</MediaLoadingMessage>
            { location === 'home' && <Home /> }
            { location === 'about' && <About loading={loading} /> }
            { location === 'work' && <Work /> }
            { location === 'contact' && <Contact /> }
          </Media>
        </Container>
      </Fragment>
    );
  },
);
