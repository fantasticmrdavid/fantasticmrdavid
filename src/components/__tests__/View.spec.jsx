import React from 'react';
import { shallow } from 'enzyme';
import View from 'components/View/ui';
import { LOCATIONS } from 'routers/constants';

const props = {
  location: LOCATIONS.about,
};

describe('Component -> View', () => {
  it('should render the View without throwing an error', () => {
    expect(shallow(<View {...props} />)).toMatchSnapshot();
  });
});
