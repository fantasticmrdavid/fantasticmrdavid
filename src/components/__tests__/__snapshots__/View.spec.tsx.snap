// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> View should render the View without throwing an error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Container-sc-1i0f3dw-0 gTMMfH"
      />
      <div
        class="styles__Container-sc-1jpprl2-0 lceywo"
      >
        <section
          class="styles__Control-sc-1jpprl2-1 kKksil"
        >
          <nav
            class="styles__Container-sc-1fdvout-0 aFnmz"
          >
            <h1
              class="styles__Title-sc-1fdvout-1 cJAooh"
            >
              Fantastic Mr David
            </h1>
            <nav
              class="styles__Nav-sc-1fdvout-3 cwEdZi"
            >
              <div
                class="styles__Container-sc-1w8m3j0-0 dmrGVg"
              >
                <div
                  class="styles__Label-sc-1w8m3j0-1 cRIjTG"
                  target="about"
                >
                  About
                </div>
                <div
                  class="styles__Content-sc-xptth5-0 gEzKKn"
                >
                  <div>
                    Meet the captain of the ship.
                  </div>
                </div>
              </div>
              <div
                class="styles__Container-sc-1w8m3j0-0 dmrGVg"
              >
                <div
                  class="styles__Label-sc-1w8m3j0-1 kLVSUR"
                  target="work"
                >
                  Latest Work
                </div>
                <div
                  class="styles__Content-sc-xptth5-0 cgeHFj"
                >
                  <div>
                    Some things I've been working on.
                  </div>
                </div>
              </div>
              <div
                class="styles__Container-sc-1w8m3j0-0 dmrGVg"
              >
                <div
                  class="styles__Label-sc-1w8m3j0-1 oFPLn"
                  target="contact"
                >
                  Contact
                </div>
                <div
                  class="styles__Content-sc-xptth5-0 cgeHFj"
                >
                  <div>
                    For relaxing times, make it Suntory time.
                  </div>
                </div>
              </div>
            </nav>
            <div
              class="styles__Subtitle-sc-1fdvout-4 fnJwRI"
            >
              <h2>
                <div
                  class="styles__CogIconWrapper-sc-1fdvout-5 gbVFgv"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                    />
                  </svg>
                </div>
                Art + Engineering
              </h2>
              <div
                class="styles__Cursor-sc-cbfjlf-0 OzODf"
              />
            </div>
            <a
              class="styles__Cta-sc-1fdvout-2 gHsBxM"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                />
              </svg>
            </a>
          </nav>
        </section>
        <section
          class="styles__Media-sc-1jpprl2-2 fuwkzu"
        >
          <div
            class="styles__Shutter-sc-1jpprl2-3 styles__ShutterTop-sc-1jpprl2-4 ddDaSX"
          />
          <div
            class="styles__Shutter-sc-1jpprl2-3 styles__ShutterBottom-sc-1jpprl2-5 dtYEwQ"
          />
          <div
            class="styles__MediaLoadingMessage-sc-1jpprl2-6 kGrthN"
          >
            Hello!
          </div>
          <div
            class="styles__Container-sc-qm8p7s-0 eoAhMa"
          >
            <div
              class="styles__Bg-sc-qm8p7s-1 bPLvTD"
            >
              <div
                class="styles__Container-sc-1pwogyu-0 fkagAV"
              >
                <div
                  class="styles__Polygon-sc-1pwogyu-1 dCIqjj"
                >
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Front-sc-1pwogyu-8 deBIOm"
                  >
                    <div
                      class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                      direction="left"
                    />
                  </div>
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Back-sc-1pwogyu-3 gCcrQb"
                  >
                    <div
                      class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                    />
                  </div>
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Top-sc-1pwogyu-6 eLxrDv"
                  />
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Bottom-sc-1pwogyu-7 cpagcj"
                  >
                    <div
                      class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                    />
                  </div>
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Left-sc-1pwogyu-5 iDSTJS"
                  >
                    <div
                      class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                      direction="left"
                    />
                  </div>
                  <div
                    class="styles__Side-sc-1pwogyu-2 styles__Right-sc-1pwogyu-4 fTRfub"
                  >
                    <div
                      class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                    />
                  </div>
                </div>
              </div>
            </div>
            <img
              alt="Me"
              class="styles__ProfilePhoto-sc-qm8p7s-2 cToiw"
              src="/assets/images/profile.jpg"
            />
            <div
              class="styles__Content-sc-qm8p7s-3 ciksqE"
            >
              <p>
                Hi! I'm David, and I like making things.
                <br />
                <br />
                Since I was a kid I've been passionate about building and creating. I've always enjoyed the journey of taking an idea from concept and making it reality.
                <br />
                <br />
                I firmly believe that by empowering people with problem solving technology and empathy we can build a better world together.
              </p>
              <h2>
                Engineer
              </h2>
              <p>
                I've been building web projects for over a decade, starting from when I was in high school and needed a simple HTML page to display my artwork. From there my journey has taken me to Full-Stack LAMP Development working on vendor-side and agency projects alike, and then to leading Front-End and UX Engineering specialising in HTML5, CSS3, ES6 JavaScript and React. I love learning about new technologies to develop better and more effective ways to solve problems and get things done.
              </p>
              <h2>
                Artist
              </h2>
              <p>
                Drawing has been my happy place for as long as I can remember. While building software keeps the puzzle solving part of my brain busy, illustration, artwork, and tinkering have allowed me to stretch my creative boundaries. In my downtime I've drawn comics, exhibited artwork, designed and produced 3D prints, and made 
                <a
                  class="styles__Link-sc-1rd8ebj-0 jONwCq styles__SleeperLink-sc-qm8p7s-6 cFShCR"
                  href="https://www.youtube.com/watch?v=7xk4LmZ6hwc"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  other random things
                </a>
                 simply because they were too much fun.
              </p>
              <h2>
                About this site
              </h2>
              fantasticmrdavid.com.au was built with:
              <div
                class="styles__SiteSpecs-sc-qm8p7s-4 hIVtxr"
              >
                <div
                  class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
                >
                  <h3>
                    Environment
                  </h3>
                  <ul>
                    <li>
                      ES6
                    </li>
                    <li>
                      Node.js
                    </li>
                    <li>
                      Webpack
                    </li>
                  </ul>
                </div>
                <div
                  class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
                >
                  <h3>
                    Architecture
                  </h3>
                  <ul>
                    <li>
                      React
                    </li>
                    <li>
                      Styled-Components
                    </li>
                  </ul>
                </div>
                <div
                  class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
                >
                  <h3>
                    Testing/QA
                  </h3>
                  <ul>
                    <li>
                      TypeScript
                    </li>
                    <li>
                      ESLint
                    </li>
                    <li>
                      Prettier
                    </li>
                    <li>
                      Jest
                    </li>
                    <li>
                      Testing Library
                    </li>
                  </ul>
                </div>
                <div
                  class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
                >
                  <h3>
                    Deployment
                  </h3>
                  <ul>
                    <li>
                      Github
                    </li>
                    <li>
                      Codeship
                    </li>
                    <li>
                      AWS
                    </li>
                  </ul>
                </div>
              </div>
              <p>
                Disable JavaScript to enable the noscript version!
              </p>
              <a
                class="styles__Link-sc-1rd8ebj-0 jONwCq"
                href="https://github.com/fantasticmrdavid/fantasticmrdavid"
                rel="noopener noreferrer"
                target="_blank"
              >
                Visit the Github repository
              </a>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Container-sc-1i0f3dw-0 gTMMfH"
    />
    <div
      class="styles__Container-sc-1jpprl2-0 lceywo"
    >
      <section
        class="styles__Control-sc-1jpprl2-1 kKksil"
      >
        <nav
          class="styles__Container-sc-1fdvout-0 aFnmz"
        >
          <h1
            class="styles__Title-sc-1fdvout-1 cJAooh"
          >
            Fantastic Mr David
          </h1>
          <nav
            class="styles__Nav-sc-1fdvout-3 cwEdZi"
          >
            <div
              class="styles__Container-sc-1w8m3j0-0 dmrGVg"
            >
              <div
                class="styles__Label-sc-1w8m3j0-1 cRIjTG"
                target="about"
              >
                About
              </div>
              <div
                class="styles__Content-sc-xptth5-0 gEzKKn"
              >
                <div>
                  Meet the captain of the ship.
                </div>
              </div>
            </div>
            <div
              class="styles__Container-sc-1w8m3j0-0 dmrGVg"
            >
              <div
                class="styles__Label-sc-1w8m3j0-1 kLVSUR"
                target="work"
              >
                Latest Work
              </div>
              <div
                class="styles__Content-sc-xptth5-0 cgeHFj"
              >
                <div>
                  Some things I've been working on.
                </div>
              </div>
            </div>
            <div
              class="styles__Container-sc-1w8m3j0-0 dmrGVg"
            >
              <div
                class="styles__Label-sc-1w8m3j0-1 oFPLn"
                target="contact"
              >
                Contact
              </div>
              <div
                class="styles__Content-sc-xptth5-0 cgeHFj"
              >
                <div>
                  For relaxing times, make it Suntory time.
                </div>
              </div>
            </div>
          </nav>
          <div
            class="styles__Subtitle-sc-1fdvout-4 fnJwRI"
          >
            <h2>
              <div
                class="styles__CogIconWrapper-sc-1fdvout-5 gbVFgv"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                  />
                </svg>
              </div>
              Art + Engineering
            </h2>
            <div
              class="styles__Cursor-sc-cbfjlf-0 OzODf"
            />
          </div>
          <a
            class="styles__Cta-sc-1fdvout-2 gHsBxM"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              />
            </svg>
          </a>
        </nav>
      </section>
      <section
        class="styles__Media-sc-1jpprl2-2 fuwkzu"
      >
        <div
          class="styles__Shutter-sc-1jpprl2-3 styles__ShutterTop-sc-1jpprl2-4 ddDaSX"
        />
        <div
          class="styles__Shutter-sc-1jpprl2-3 styles__ShutterBottom-sc-1jpprl2-5 dtYEwQ"
        />
        <div
          class="styles__MediaLoadingMessage-sc-1jpprl2-6 kGrthN"
        >
          Hello!
        </div>
        <div
          class="styles__Container-sc-qm8p7s-0 eoAhMa"
        >
          <div
            class="styles__Bg-sc-qm8p7s-1 bPLvTD"
          >
            <div
              class="styles__Container-sc-1pwogyu-0 fkagAV"
            >
              <div
                class="styles__Polygon-sc-1pwogyu-1 dCIqjj"
              >
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Front-sc-1pwogyu-8 deBIOm"
                >
                  <div
                    class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                    direction="left"
                  />
                </div>
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Back-sc-1pwogyu-3 gCcrQb"
                >
                  <div
                    class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                  />
                </div>
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Top-sc-1pwogyu-6 eLxrDv"
                />
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Bottom-sc-1pwogyu-7 cpagcj"
                >
                  <div
                    class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                  />
                </div>
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Left-sc-1pwogyu-5 iDSTJS"
                >
                  <div
                    class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                    direction="left"
                  />
                </div>
                <div
                  class="styles__Side-sc-1pwogyu-2 styles__Right-sc-1pwogyu-4 fTRfub"
                >
                  <div
                    class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                  />
                </div>
              </div>
            </div>
          </div>
          <img
            alt="Me"
            class="styles__ProfilePhoto-sc-qm8p7s-2 cToiw"
            src="/assets/images/profile.jpg"
          />
          <div
            class="styles__Content-sc-qm8p7s-3 ciksqE"
          >
            <p>
              Hi! I'm David, and I like making things.
              <br />
              <br />
              Since I was a kid I've been passionate about building and creating. I've always enjoyed the journey of taking an idea from concept and making it reality.
              <br />
              <br />
              I firmly believe that by empowering people with problem solving technology and empathy we can build a better world together.
            </p>
            <h2>
              Engineer
            </h2>
            <p>
              I've been building web projects for over a decade, starting from when I was in high school and needed a simple HTML page to display my artwork. From there my journey has taken me to Full-Stack LAMP Development working on vendor-side and agency projects alike, and then to leading Front-End and UX Engineering specialising in HTML5, CSS3, ES6 JavaScript and React. I love learning about new technologies to develop better and more effective ways to solve problems and get things done.
            </p>
            <h2>
              Artist
            </h2>
            <p>
              Drawing has been my happy place for as long as I can remember. While building software keeps the puzzle solving part of my brain busy, illustration, artwork, and tinkering have allowed me to stretch my creative boundaries. In my downtime I've drawn comics, exhibited artwork, designed and produced 3D prints, and made 
              <a
                class="styles__Link-sc-1rd8ebj-0 jONwCq styles__SleeperLink-sc-qm8p7s-6 cFShCR"
                href="https://www.youtube.com/watch?v=7xk4LmZ6hwc"
                rel="noopener noreferrer"
                target="_blank"
              >
                other random things
              </a>
               simply because they were too much fun.
            </p>
            <h2>
              About this site
            </h2>
            fantasticmrdavid.com.au was built with:
            <div
              class="styles__SiteSpecs-sc-qm8p7s-4 hIVtxr"
            >
              <div
                class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
              >
                <h3>
                  Environment
                </h3>
                <ul>
                  <li>
                    ES6
                  </li>
                  <li>
                    Node.js
                  </li>
                  <li>
                    Webpack
                  </li>
                </ul>
              </div>
              <div
                class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
              >
                <h3>
                  Architecture
                </h3>
                <ul>
                  <li>
                    React
                  </li>
                  <li>
                    Styled-Components
                  </li>
                </ul>
              </div>
              <div
                class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
              >
                <h3>
                  Testing/QA
                </h3>
                <ul>
                  <li>
                    TypeScript
                  </li>
                  <li>
                    ESLint
                  </li>
                  <li>
                    Prettier
                  </li>
                  <li>
                    Jest
                  </li>
                  <li>
                    Testing Library
                  </li>
                </ul>
              </div>
              <div
                class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
              >
                <h3>
                  Deployment
                </h3>
                <ul>
                  <li>
                    Github
                  </li>
                  <li>
                    Codeship
                  </li>
                  <li>
                    AWS
                  </li>
                </ul>
              </div>
            </div>
            <p>
              Disable JavaScript to enable the noscript version!
            </p>
            <a
              class="styles__Link-sc-1rd8ebj-0 jONwCq"
              href="https://github.com/fantasticmrdavid/fantasticmrdavid"
              rel="noopener noreferrer"
              target="_blank"
            >
              Visit the Github repository
            </a>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
